File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int left = 0, right = 0;
        int jump = 0;

        while (right < nums.length - 1) {
            int farthest = 0;
            for(int i = left; i < right + 1; i++) {
                farthest = Math.max(farthest, i + nums[i]);
            }
            left = right + 1;
            right = farthest;            
            jump++;
        }

        return jump;
    }
} 

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        l = r = res = 0

        while r < len(nums) - 1: # len(nums) - 1 is the target
            farthest = 0
            for i in range(l, r + 1): # l to r because the last element in the range is not inclusive
                farthest = max(farthest, i + nums[i])
            l = r + 1
            r = farthest
            res += 1
        
        return res